File: model/src/main/java/com/github/fburato/highwheelmodules/model/classpath/AccessVisitor.java
package com.github.fburato.highwheelmodules.model.classpath;

import com.github.fburato.highwheelmodules.model.bytecode.AccessPoint;
import com.github.fburato.highwheelmodules.model.bytecode.AccessType;
import com.github.fburato.highwheelmodules.model.bytecode.ElementName;

public interface AccessVisitor {

    void apply(AccessPoint source, AccessPoint dest, AccessType type);

    void newNode(ElementName clazz);

    void newAccessPoint(AccessPoint ap);

    void newEntryPoint(ElementName clazz);

}

File: model/src/main/scala/com/github/fburato/highwheelmodules/model/classpath/AccessVisitor.scala
package com.github.fburato.highwheelmodules.model.classpath

import com.github.fburato.highwheelmodules.model.bytecode.{AccessPoint, AccessType, ElementName}

trait AccessVisitor {
  def apply(source: AccessPoint, dest: AccessPoint, accessType: AccessType): Unit

  def newNode(clazz: ElementName): Unit

  def newAccessPoint(ap: AccessPoint): Unit

  def newEntryPoint(clazz: ElementName): Unit
}
